server/src-lib/Hasura/GraphQL/Execute/LiveQuery.hs

```diff
@@ -161,33 +167,56 @@ type TextEncodedVariables
 -- referring correctly to the values from '_subs' temporary table
 -- The variables are at _subs.result_vars.variables and
 -- session variables at _subs.result_vars.user
-toMultiplexedQueryVar
-  :: (MonadState GV.AnnPGVarVals m)
-  => GR.UnresolvedVal -> m S.SQLExp
-toMultiplexedQueryVar = \case
-  GR.UVPG annPGVal ->
-    let GR.AnnPGVal varM isNullable colTy colVal = annPGVal
-    in case (varM, isNullable) of
-      -- we don't check for nullability as
-      -- this is only used for reusable plans
-      -- the check has to be made before this
-      (Just var, _) -> do
-        modify $ Map.insert var (colTy, colVal)
-        return $ fromResVars (PgTypeSimple colTy)
-          [ "variables"
-          , G.unName $ G.unVariable var
-          ]
-      _             -> return $ toTxtValue colTy colVal
-  GR.UVSessVar ty sessVar ->
-    return $ fromResVars ty [ "user", T.toLower sessVar]
-  GR.UVSQL sqlExp -> return sqlExp
+type FallbackOpPartial = (GR.QueryRootFldUnresolved, Set.HashSet G.Variable)
```

Comment:
Minor, mostly unimportant comment, but: I wonder if there’s a good way to encode into the type system that this `HashSet` must be non-empty.